yew/src/virtual_dom/vnode.rs

```diff
@@ -65,14 +75,15 @@ impl VNode {
         }
     }
 
-    pub fn key(&self) -> &Option<String> {
+    pub(crate) fn move_before(&self, parent: &Element, next_sibling: Option<Node>) {
         match self {
-            VNode::VTag(vtag) => &vtag.key,
-            VNode::VText(_) => &None,
-            VNode::VComp(vcomp) => &vcomp.key,
-            VNode::VList(vlist) => &vlist.key,
-            VNode::VRef(_) => &None,
-        }
+            VNode::VList(vlist) => {
+                for node in vlist.children.iter() {
+                    node.move_before(parent, next_sibling.clone());
+                }
+            }
+            _ => super::insert_node(&self.first_node(), parent, next_sibling),
```

Comment:
I think we need to recurse here because `VComp` could have a `VList` as its root node

Reply:
I removed both, and I will re-use `diff_layouts` only, it's convenient :) (but indeed, looks like I removed the wrong one in this last commit).

Reply:
I didn't think of this case, thanks :)

Reply:
@jstarry How can I get all the child nodes for a VComp? There is only one NodeRef.

Reply:
Just pushed a test that exhibits that, but I don't how to fix. I confess that VComp stay a mystery for me ^^

Reply:
> @jstarry How can I get all the child nodes for a VComp? There is only one NodeRef.

We need the `last_root` from `ComponentState` but that's tough because `VComp` only has an untyped `AnyScope`. I'll play around with this today. If we can pull out `last_root`, then it's just a matter of recursion

Reply:
@totorigolo I opened [this PR](https://github.com/yewstack/yew/pull/1333) which will add a new `VComp::root_vnode()` method which you can use here